messaging/src/main/java/org/axonframework/common/ProcessUtils.java

```diff
@@ -49,4 +51,30 @@ public static void executeWithRetry(Runnable runnable, Predicate<RuntimeExceptio
 
         throw lastException;
     }
+
+    /**
+     * Executes an action, with potential retry in case the result is false. Exception handling should be taken care of
+     * within the action if needed.
+     *
+     * @param runnable      action to execute
```

Comment:
Nit: I would add a similar line as the main description, stating this runnable is leading to decide if a retry occurs.